### PR TITLE
Bypass gRPC stack when the search service is local

### PR DESCRIPTION
**Pull request by guilload**
### Description
Per title. I'm doing so by refactoring how the search job placer accesses the available list of searcher nodes/clients.

### How was this PR tested?
- [x] Updated unit tests
- [x] Re-enable some tests
- [x] Ran `make test-all`
- [x] Manually launched two nodes locally and issued some basic search queries

### CODE REVIEW

**Comment by guilload** [comment]
@fmassot, this is ready for review, I just have to re-enable a few tests, and it should be good to go.

File: quickwit/quickwit-common/src/tower/pool.rs [hunk 1]
```diff
@@ -32,7 +32,6 @@ use super::Change;
 
 /// A pool of `S` services identified by `K` keys. The pool can be updated manually by calling the
```
Comment (fmassot) [inline quickwit/quickwit-common/src/tower/pool.rs:33]: Nitpick: not necessary `services`, just values.

File: quickwit/quickwit-search/src/retry/mod.rs [hunk 2]
```diff
@@ -59,34 +60,36 @@ impl<'a> Job for &'a str {
...
 
-    fn cost(&self) -> u32 {
+    fn cost(&self) -> usize {
         1
     }
 }
 
 // Select a new client from the client pool by the following oversimplified policy:
 // 1. Take the first split_id of the request
 // 2. Ask for a relevant client for that split while excluding the failing client.
```
Comment (fmassot) [inline quickwit/quickwit-search/src/retry/mod.rs:70]:
```suggestion
// 2. Ask for a relevant client for that split while excluding the given `excluded_addr`.
```

File: quickwit/quickwit-search/src/search_job_placer.rs [hunk 3]
```diff
@@ -17,283 +17,240 @@
...
-}
-
 /// Node is a utility struct used to represent a rendez-vous hashing node.
 /// It's used to track the load and the computed hash for a given key
-#[derive(Clone, Debug)]
-struct Node {
-    pub peer_grpc_addr: SocketAddr,
-    pub load: u64,
+#[derive(Debug, Clone)]
+struct JobCandidate {
```
Comment (fmassot) [inline quickwit/quickwit-search/src/search_job_placer.rs:72]: `NodeCandidate` sounds more natural here.

Comment (fmassot) [reply quickwit/quickwit-search/src/search_job_placer.rs:72]: as we assign jobs to node

File: quickwit/quickwit-serve/src/lib.rs [hunk 4]
```diff
@@ -467,6 +466,53 @@ where R: Rate
     }
 }
 
+async fn setup_searcher(
```
Comment (fmassot) [inline quickwit/quickwit-serve/src/lib.rs:469]: is there a test on this setup?

**Review by fmassot — COMMENTED** [review]
LGTM. A test on `setup_searcher` would be welcomed.

**Review by fmassot — APPROVED** [review]

**Comment by guilload** [comment]
I wrote a really nice unit test for you and my future self :)

**Pull request merged**